### ISSUE TITLE
geom_boxplot + width != 1 moves boxes

### ISSUE THREAD

**Issue by ldecicco-USGS** [tidyverse/ggplot2#2331]
(possibly related to #2290)
I'd like to make the width of the boxplots a bit fatter, but when I do that, the labels no longer align with the boxplot:

```r
ggplot(data = starwars) + 
geom_boxplot(aes(x = hair_color, y = mass), width = 2) + 
scale_y_log10()
```
![image](https://user-images.githubusercontent.com/1105215/32614335-a4e9980c-c532-11e7-915b-b61b3ba078b1.png)

I would have guessed the width of the boxplot gets bigger, but the position would stay the same.

**Comment by ldecicco-USGS** [comment]
I should mention, adding 
```r
position = position_dodge2(preserve = "total")
```
as described in #2290 eliminated my *need* to widen up the box plots (I'm doing a bunch of facet boxplots)...but this behavior still seems strange to me.

**Comment by hadley** [comment]
Reprex with simpler data, although it needs a bigger width before the problem becomes apparent:

``` r
library(ggplot2)
df <- data.frame(
  x = rep(c("a", "b"), each = 3),
  y = c(1, 2, 3, 5, 6, 7)
)

ggplot(df, aes(x, y)) + 
  geom_boxplot(width = 1) +
  geom_vline(xintercept = c(0.5, 1, 1.5, 2, 2.5), alpha = 1/4) 
```

![](https://i.imgur.com/BeDaCEe.png)

``` r
  
ggplot(df, aes(x, y)) + 
  geom_boxplot(width = 2) +
  geom_vline(xintercept = c(0.5, 1, 1.5, 2, 2.5), alpha = 1/4) 
```

![](https://i.imgur.com/rGShhmA.png)

``` r

ggplot(df, aes(x, y)) + 
  geom_boxplot(width = 3) +
  geom_vline(xintercept = c(0.5, 1, 1.5, 2, 2.5), alpha = 1/4) 
```

![](https://i.imgur.com/PdV3Z7F.png)

I bet this is because boxplots now are wide enough that they overlap, and hence the new dodging algorithm combines into a single group.

@karawoo maybe this is correct behaviour and not actually a bug?

**Comment by karawoo** [comment]
`position_dodge2()` actually considers both the second and third overlapping. I _think_ the reason that the third plot looks messed up but the second doesn't is these lines:

https://github.com/tidyverse/ggplot2/blob/064873910b164ebab555990343859db1044b7d64/R/position-dodge2.r#L83-L86

In the third width is 3 and number of elements is 2, so after this calculation the width is still greater than 1, whereas in the second it's exactly 1.

I think `position_dodge2()` is behaving correctly. But, I'm not entirely sure what behavior users expect when asking for `geom_boxplot(width = [something >1])`. I can think of two possibilities:

1. The boxes should overlap each other. This can be done with `position_dodge()` (though it throws a warning). Avoiding overlap was part of the motivation for `position_dodge2()` in the first place, so I'm not sure how to change it to support overlap.
2. The boxes should just appear wider on the screen, and corresponding axis ticks etc. should also be farther apart. This is so dependent on the size and aspect ratio of the plotting window anyway that to me it makes more sense to change those rather than the plotting data. 

tl;dr I think the current behavior is correct. But maybe we should document somewhere that width should not be greater than 1?

**Comment by hadley** [comment]
I agree - I think this is correct behaviour.

I'm not sure how we should document since width could be greater than one if the boxplots are spread further apart (i.e. numeric x rather than categorical x). So I think we're ok here.
